packages/core/src/telemetry/emitter.ts

```diff
@@ -82,29 +88,46 @@ export const telemetryEmitter = async ({ dataDir, version, debug }: {
     run_id: runId,
   }
 
-  const identify: IdentifyFunction = (...args) => {
-    const [id, person] = (
-      typeof args[0] === 'string' ? args : [undefined, args[0]]
-    ) as [string, TelemetryProperties | undefined] | [undefined, TelemetryProperties]
-
-    const isCurrentIdAnonymous = currentId === runId
+  const identify: IdentifyFunction = (...args: unknown[]) => {
+    let groupIdentityType: GroupIdentityType | undefined; let groupId: string | undefined; let
+      props: TelemetryProperties
+    if (typeof args[2] === undefined && typeof args[1] === undefined) {
+      props = args[0] as TelemetryProperties
+    } else if (typeof args[2] === undefined) {
+      groupIdentityType = args[0] as GroupIdentityType
+      groupId = groupIdentities[groupIdentityType]
+      props = args[1] as TelemetryProperties
+    } else {
+      [groupIdentityType, groupId, props] = args as [GroupIdentityType, string, TelemetryProperties ]
+      groupIdentities[groupIdentityType] = groupId
+    }
 
-    if (isCurrentIdAnonymous || Object.keys(person ?? {}).length) {
+    if (groupIdentityType && groupId) {
+      pushEvent({
+        event: '$groupidentify',
```

Comment:
https://posthog.com/docs/getting-started/group-analytics#the-difference-between-groups-and-cohorts:

```
If your only goal is to create a list of users with something in common, we recommend cohorts instead of groups.
```

Reply:
Using a group makes sense in it because we want to aggregate events based on profile.